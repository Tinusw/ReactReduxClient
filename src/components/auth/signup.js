import React, { Component } from "react";
import { connect } from "react-redux";
import * as actions from "../../actions";

export class SignUp extends Component {
  constructor() {
    super();
    this.state = {
      email: "",
      password: "",
      passwordConfirm: ""
    };
  }

  shouldComponentUpdate(nextProps) {
    if (nextProps.authenticated === true) {
      this.props.history.push("/campaign/index");
    }
    return true;
  }

  handleInputChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  handleSubmit = event => {
    event.preventDefault();
    this.props.signUpUser(this.state);
  };

  renderError() {
    if (
      this.state.passwordConfirm !== "" &&
      this.state.password !== "" &&
      this.state.password !== this.state.passwordConfirm
    ) {
      return (
        <div className="error" id="error">
          passwords do not match :(
        </div>
      );
    }
  }

  render() {
    return (
      <form className="SignUp" onSubmit={this.handleSubmit}>
        <label>email</label>
        <input
          id="email"
          onChange={this.handleInputChange}
          name="email"
          type="text"
          autoComplete="email"
        />
        <label>password</label>
        <input
          id="password"
          onChange={this.handleInputChange}
          name="password"
          type="password"
        />
        <label>confirm password</label>
        <input
          id="passwordConfirm"
          onChange={this.handleInputChange}
          name="passwordConfirm"
          type="password"
        />
        <button>Sign Up</button>
        {this.renderError()}
      </form>
    );
  }
}

function mapStateToProps(state) {
  return {
    errorMessage: state.auth.error,
    authenticated: state.auth.authenticated
  };
}

export default connect(mapStateToProps, actions)(SignUp);
