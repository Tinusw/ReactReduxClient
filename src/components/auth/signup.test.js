import React from "react";
import * as test_tools from "../../testing/test_helper.js";

import { SignUp } from "./signup";

describe("<signUp/>", () => {
  const mockSignUpfn = jest.fn();

  let actual;

  beforeEach(() => {
    actual = test_tools.shallow(<SignUp signUpUser={mockSignUpfn} />);
  });

  it("renders correctly", () => {
    expect(actual).toMatchSnapshot();
  });

  it("should have the className SignUp", () => {
    expect(actual.find("form.SignUp").length).toBe(1);
  });

  it("should call signinUser on submit", () => {
    actual.find("form").simulate("submit", { preventDefault() {} });
    expect(mockSignUpfn.mock.calls.length).toBe(1);
  });

  it("should call signinUser with the email and password in the state as arguments", () => {
    actual.find("#email").simulate("change", {
      target: { name: "email", value: "blah@gmail.com" }
    });
    actual.find("#password").simulate("change", {
      target: { name: "password", value: "1234" }
    });
    actual.find("form").simulate("submit", { preventDefault() {} });
    expect(mockSignUpfn.mock.calls[1][0]).toEqual({
      email: "blah@gmail.com",
      password: "1234"
    });
  });
});
