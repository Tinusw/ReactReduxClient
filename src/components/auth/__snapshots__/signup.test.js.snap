// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<signUp/> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp
    signUpUser={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label>
          email
</label>,
        <input
          autoComplete="email"
          id="email"
          name="email"
          onChange={[Function]}
          type="text"
/>,
        <label>
          password
</label>,
        <input
          id="password"
          name="password"
          onChange={[Function]}
          type="password"
/>,
        <label>
          confirm password
</label>,
        <input
          id="passwordConfirm"
          name="passwordConfirm"
          onChange={[Function]}
          type="password"
/>,
        <button>
          Sign Up
</button>,
        undefined,
      ],
      "className": "SignUp",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "email",
        },
        "ref": null,
        "rendered": "email",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoComplete": "email",
          "id": "email",
          "name": "email",
          "onChange": [Function],
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "password",
        },
        "ref": null,
        "rendered": "password",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "password",
          "name": "password",
          "onChange": [Function],
          "type": "password",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "confirm password",
        },
        "ref": null,
        "rendered": "confirm password",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "passwordConfirm",
          "name": "passwordConfirm",
          "onChange": [Function],
          "type": "password",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign Up",
        },
        "ref": null,
        "rendered": "Sign Up",
        "type": "button",
      },
      undefined,
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            email
</label>,
          <input
            autoComplete="email"
            id="email"
            name="email"
            onChange={[Function]}
            type="text"
/>,
          <label>
            password
</label>,
          <input
            id="password"
            name="password"
            onChange={[Function]}
            type="password"
/>,
          <label>
            confirm password
</label>,
          <input
            id="passwordConfirm"
            name="passwordConfirm"
            onChange={[Function]}
            type="password"
/>,
          <button>
            Sign Up
</button>,
          undefined,
        ],
        "className": "SignUp",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "email",
          },
          "ref": null,
          "rendered": "email",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoComplete": "email",
            "id": "email",
            "name": "email",
            "onChange": [Function],
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "password",
          },
          "ref": null,
          "rendered": "password",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "password",
            "name": "password",
            "onChange": [Function],
            "type": "password",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "confirm password",
          },
          "ref": null,
          "rendered": "confirm password",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "passwordConfirm",
            "name": "passwordConfirm",
            "onChange": [Function],
            "type": "password",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "button",
        },
        undefined,
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
